docker/reference/normalize.go

```diff
@@ -29,11 +29,18 @@ type normalizedNamed interface {
 // transforming a familiar name from Docker UI to a fully
 // qualified reference. If the value may be an identifier
 // use ParseAnyReference.
-func ParseNormalizedNamed(s string) (Named, error) {
+func ParseNormalizedNamed(sys *types.SystemContext, s string) (Named, error) {
 	if ok := anchoredIdentifierRegexp.MatchString(s); ok {
 		return nil, fmt.Errorf("invalid repository name (%s), cannot specify 64-byte hexadecimal strings", s)
 	}
-	domain, remainder := splitDockerDomain(s)
+	registries, err := GetRegistries(sys)
+	if err != nil {
+		return nil, err
+	}
+	if len(registries) == 0 {
```

Comment:
Should we default to quay.io in this case?  Or fedora?

Reply:
I really don't think the containers/image should have a default.  In the perfect world we would just allow a "" as the registry.

Reply:
A minimal comment for now:

`c/i/docker/reference.Named`, `!IsNameOnly`, is a type representing a _specific_ image in a _specific_ registry.

Maybe there needs to be a type representing the “unqualified” inputs to search.

Maybe some uses of `reference.Named` should not use it at all, and accept more lax input. (In particular I was wrong about `containers-storage:`, it uses `reference.ParseNormalizedNamed` and does not allow hostname-less references.)

But, if hostname-qualified and hostname-less references should live side by side, they _need_ to be distinct types. The experience, _both_ with projectatomic/docker’s `reference`, and with podman’s `string` uses, is that mixing them together makes it all to easy to be careless about the semantics and results in surprising and unexpected corner case behaviors.